### PR TITLE
finetuning with Megatron  + HF lm

### PR DESCRIPTION
**Pull request by ekmb**
- [x] [BioMegatron models added and will be automatically downloaded with NeMo

- [x] tutorial on how to use Megatron-LM

- [x] section on how to use HF lm models

https://colab.research.google.com/drive/1HLqyPelWTVyh5ZQugVqwMY5LmiENby-s?authuser=3#scrollTo=Wi_LBdSA4U6Q

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -172,18 +172,21 @@ To run tutorials:
...
+    * - NLP
+     - Using Pretrained Language Models for Downstream Tasks
+     - `01_Pretrained_Language_Models_for_Downstream_Tasks.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/01_Pretrained_Language_Models_for_Downstream_Tasks.ipynb>`_
+    * - NLP
+     - Question answering with SQuAD
+     - `02_Question_Answering_Squad.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/02_Question_Answering_Squad.ipynb>`_
+    * - NLP
      - Token Classification (Named Entity Recognition)
-     - `Token_Classification_Named_Entity_Recognition.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/Token_Classification_Named_Entity_Recognition.ipynb>`_
+     - `03_Token_Classification_Named_Entity_Recognition.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/03_Token_Classification_Named_Entity_Recognition.ipynb>`_
```
Comment (yzhang123) [inline README.rst:183]: can you postpone this renaming of existing file, i will get bad conflicts.

File: README.rst [hunk 2]
```diff
@@ -172,18 +172,21 @@ To run tutorials:
...
    * - NLP
      - GLUE Benchmark
-     - `GLUE_Benchmark.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/GLUE_Benchmark.ipynb>`_
+     - `04_GLUE_Benchmark.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/04_GLUE_Benchmark.ipynb>`_
```
Comment (yzhang123) [inline README.rst:186]: same here

File: README.rst [hunk 3]
```diff
@@ -172,18 +172,21 @@ To run tutorials:
...
    * - NLP
      - Punctuation and Capitialization
-     - `Punctuation_and_Capitalization.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/Punctuation_and_Capitalization.ipynb>`_
-   * - NLP
-     - Question answering with SQuAD
-     - `Question_Answering_Squad.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/Question_Answering_Squad.ipynb>`_
+     - `05_Punctuation_and_Capitalization.ipynb <https://colab.research.google.com/github/NVIDIA/NeMo/blob/main/tutorials/nlp/05_Punctuation_and_Capitalization.ipynb>`_
```
Comment (yzhang123) [inline README.rst:189]: same here

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py [hunk 4]
```diff
@@ -68,33 +81,38 @@ def get_megatron_lm_model(
...
     checkpoint_file: Optional[str] = None,
-):
-    '''
+) -> Tuple[MegatronBertEncoder, str]:
+    """
     Returns the dict of special tokens associated with the model.
+
     Args:
-        pretrained_mode_name ('str'): name of the pretrained model from the hugging face list,
+        pretrained_mode_name: name of the pretrained model from the hugging face list,
```
Comment (yzhang123) [inline nemo/collections/nlp/modules/common/megatron/megatron_utils.py:89]: this is megatron, so not huggingface list

**Review by yzhang123 — CHANGES_REQUESTED** [review]

**Review by okuchaiev — APPROVED** [review]
looks good to me

**Review by yzhang123 — COMMENTED** [review]
I have not checked notebooks so i cannot approve it formally, but Oleksii has approved, so he can merge it since he has admin rights. I'm ok with that given he has reviewed the notebooks
Rest looks good to me

**Comment by ekmb** [comment]
> I have not checked notebooks so i cannot approve it formally, but Oleksii has approved, so he can merge it since he has admin rights. I'm ok with that given he has reviewed the notebooks
> Rest looks good to me

@okuchaiev up to you what to do with the PR

**Review by yzhang123 — APPROVED** [review]
same comment as above, fine given Oleksii reviewed the notebook

**Pull request merged**